Source/Public/Get-VSTeamUserEntitlement.ps1

```diff
@@ -94,21 +95,34 @@ function Get-VSTeamUserEntitlement {
                }
                $filter = $filter.SubString(0, $filter.Length - 5)
             }
-            $listurl += _appendQueryString -name "`$filter" -value $filter
-            $listurl += _appendQueryString -name "select" -value ($select -join ",")
+            if($filter){
+                $qs.Add("`$filter", $filter)
+            }
+            if($select) {
+                $qs.Add("select", $select -join ",")
+            }
 
             # Call the REST API
+            # TODO: will break with blank API version
+            if($qs.HasKeys()){
+                $listurl += "&" + $qs.ToString()
+            }
+
             Write-Verbose "API params: $listurl"
             $items = _callAPIContinuationToken -Url $listurl -PropertyName "members"
             foreach ($item in $items) {
                $objs += [vsteam_lib.UserEntitlement]::new($item)
             }
          } else {
-            $listurl += _appendQueryString -name "top" -value $top -retainZero
-            $listurl += _appendQueryString -name "skip" -value $skip -retainZero
-            $listurl += _appendQueryString -name "select" -value ($select -join ",")
+            $qs.Add("top", $top)
+            $qs.Add("skip", $skip)
+            if($select) {
+               $qs.Add("select", $select -join ",")
+            }
 
             # Call the REST API
+            # TODO: will break with blank API version
+            $listurl += "&" + $qs.ToString()

```

Comment:
What do you think has to be done here or where exactly?

Reply:
For correctness? Same thing I did elsewhere:
```
if($listurl.IndexOf('?') -gt 0) {
    $listurl += "&" + $qs.ToString()
}
else {
    $listurl += "?" + $qs.ToString()
}
```